src/Servers/Kestrel/Core/src/Internal/Http2/Http2MessageBody.cs

```diff
@@ -19,14 +19,15 @@ public Http2MessageBody(Http2Stream context)
         : base(context)
     {
         _context = context;
+        ExtendedConnect = _context.IsExtendedConnectRequest;
     }
 
     protected override void OnReadStarting()
     {
         // Note ContentLength or MaxRequestBodySize may be null
         var maxRequestBodySize = _context.MaxRequestBodySize;
 
-        if (_context.RequestHeaders.ContentLength > maxRequestBodySize)
+        if (!ExtendedConnect && _context.RequestHeaders.ContentLength > maxRequestBodySize)
```

Comment:
```suggestion
        if (_context.RequestHeaders.ContentLength > maxRequestBodySize)
```

Do we even allow `Content-Length` request headers for extended CONNECTs? I don't want to add checks for things that cannot happen. I cannot imagine not wanting to reject a `Content-Length` request headers greater than the `MaxRequestBodySize` in any case.

Reply:
You're right that it would be weird to even have a Content-Length, so we don't have to check this here.